x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -834,7 +835,18 @@ protected void doSaveState(IndexerState indexerState, DataFrameIndexerPosition p
                                     onStop();
                                     transformTask.shutdown();
                                 }
-                                next.run();
+                                transformsConfigManager.deleteOldTransformStoredDocuments(transformId, ActionListener.wrap(
```

Comment:
Will this be called everytime `doSaveState` is called? Is it possible to add something to the response to say whether or not the deletion is necessary